Mage.Sets/src/mage/cards/a/AlaniaDivergentStorm.java

```diff
@@ -0,0 +1,190 @@
+                AlaniaDivergentStormCondition.instance, ""
+        );
+        ability.addWatcher(new AlaniaDivergentStormWatcher());
+        this.addAbility(ability);
+
+    }
+
+    private AlaniaDivergentStorm(final AlaniaDivergentStorm card) {
+        super(card);
+    }
+
+    @Override
+    public AlaniaDivergentStorm copy() {
+        return new AlaniaDivergentStorm(this);
+    }
+}
+
+// Based on MarathWillOfTheWildRemoveCountersCost
+class AlaniaDivergentStormCost extends CostImpl {
+
+    AlaniaDivergentStormCost() {
+        this.text = "have target opponent draw a card";
+        this.addTarget(new TargetOpponent());
+    }
+
+    private AlaniaDivergentStormCost(AlaniaDivergentStormCost cost) {
+        super(cost);
+    }
+
+    @Override
+    public boolean canPay(Ability ability, Ability source, UUID controllerId, Game game) {
+        Player player = game.getPlayer(controllerId);
+        if (player == null) {
+            return false;
+        }
+        for (UUID opponentID : game.getOpponents(controllerId)){
+            Player opponent = game.getPlayer(opponentID);
+            if (opponent == null) {
+                continue;
+            }
+            if (opponent.canBeTargetedBy(source.getSourceObject(game), controllerId, source, game)) {
+                return true;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean pay(Ability ability, Game game, Ability source, UUID controllerId, boolean noMana, Cost costToPay) {
+        this.getTargets().clearChosen();
+        paid = false;
+        if (this.getTargets().choose(Outcome.DrawCard, controllerId, source.getSourceId(), source, game)) {
+            Player opponent = game.getPlayer(this.getTargets().get(0).getTargets().get(0));
+            if (opponent == null || !opponent.canRespond()){
+                return false;
+            }
+            paid = opponent.drawCards(1, source, game) != 0;
+        }
+        return paid;
+    }
+
+    @Override
+    public AlaniaDivergentStormCost copy() {
+        return new AlaniaDivergentStormCost(this);
+    }
+}
+
+// Based on MastermindPlumCondition
+enum AlaniaDivergentStormCondition implements Condition {
+    instance;
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Spell spell = (Spell) source.getEffects().get(0).getValue("spellCast");
+        if (spell == null) {
+            return false;
+        }
+        AlaniaDivergentStormWatcher watcher = game.getState().getWatcher(AlaniaDivergentStormWatcher.class);
+        return spell.getId().equals(watcher.getIdOfFirstInstantCast(source.getControllerId())) ||
+                spell.getId().equals(watcher.getIdOfFirstSorceryCast(source.getControllerId())) ||
+                spell.getId().equals(watcher.getIdOfFirstOtterCast(source.getControllerId()));
+    }
+}
+
+// Based on FirstSpellCastThisTurnWatcher
```

Comment:
well the reference watcher is bugged, because it's checking id without zcc, needs to save MageObjectReference instead.